File: articles/the-history-of-programming-languages/cpp/language_trait_summary.cpp
#include <iostream>
#include <map>
#include <string>

int main() {
    std::map<std::string, std::string> languages = {
        {"Fortran", "scientific numerical programming"},
        {"Lisp", "symbolic computation"},
        {"SQL", "declarative data querying"},
        {"Rust", "memory-safe systems programming"}
    };
    for (const auto& item : languages) {
        std::cout << item.first << ": " << item.second << "\n";
    }
}

File: articles/the-history-of-programming-languages/java/LanguageTraitSummary.java
import java.util.LinkedHashMap;
import java.util.Map;

public class LanguageTraitSummary {
    public static void main(String[] args) {
        Map<String, String> languages = new LinkedHashMap<>();
        languages.put("Fortran", "scientific numerical programming");
        languages.put("Lisp", "symbolic computation");
        languages.put("SQL", "declarative data querying");
        languages.put("Rust", "memory-safe systems programming");
        for (Map.Entry<String, String> entry : languages.entrySet()) {
            System.out.printf("%s: %s%n", entry.getKey(), entry.getValue());
        }
    }
}
